src/_guides/libraries/testing.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: "Dart Testing"
+description: "How to test Dart, Flutter, and 'Dart for the Web' applications?"
+---
+
+Software testing, an important part of app development, helps verify that
+your app is working correctly before unleashing it on your users.
+This Dart testing guide outlines several types of testing, and points
+you to where you can learn more about testing on the Flutter, Dart VM,
+and 'Dart for the Web' platforms.
+
+## Testing types
+
+The Dart testing docs focus on three types of testing, out of the
+[many testing types](http://www.aptest.com/testtypes.html) that
+you may be familiar with: unit, widget (or component, depending on the
+platform), and integration (or end-to-end). Testing terminology varies,
+but these are the terms and concepts that you are likely to
+encounter when using Dart technologies:
+
+* _Unit_ tests focus on verifying the smallest piece of testable
+  software, such as a function, method, or class. Your testing suite
+  should have more unit tests that widget or integration tests.
+
+* _Widget_ (or _component_) tests verify that a widget (which
+  usually consists of multiple classes) behaves as expected.
+  A widget test requires a mock framework that can mimic user actions,
+  events, perform layout, and instantiate child widgets.
+  Note that Flutter refers to widgets, and AngularDart refers to
+  components.
+
+* _Integration_ or (_end-to-end_) tests verify the behavior of
+  an entire app, or a large chunk of an app. An integration test
+  generally runs on a real device or an OS simulator, and consists
+  of two pieces: the app itself, and the test app that puts it through
+  its paces. An integration test often measures performance, so the
+  test app generally runs on a different device or OS.
+
+## Generally useful libraries
+
+How you write your tests partly depends on whether you are using Flutter,
+Dart VM, AngularDart, or other web solutions, but the following packages
+are useful across Dart platforms:
+
+* [package:test](https://pub.dartlang.org/packages/test)<br>
+  Provides a standard way of writing tests in Dart. You can use the test
+  package to:
+    * Write single tests, or groups of tests.
+    * Use the `@TestOn` annotation to restrict tests to run on
+      specific environments.
+    * Write asynchronous tests just as you would for non-asynchronous
+      tests.
+    * Tag tests using the `@Tag` annotation. For example, define a tag to
+      create a custom configuration for some tests, or to identify some tests
+      as needing more time to complete.
+    * Create a `dart_test.yaml` file to configure tagged tests across
+      multiple files or an entire package.
+
+
+* [package:mockito](https://pub.dartlang.org/packages/mockito)<br>
+  Provides a way to mock interfaces so that a piece of code is replaced
+  by a dummy implementation that emulates real code, but behaves in
```

Comment:
How about something like:

---
Provides a way to create [mock objects](https://en.wikipedia.org/wiki/Mock_object), easily configured for use in fixed scenarios, and to verify that the system under test interacts with the mock object in expected ways.

---